axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/MrpServiceImpl.java

```diff
@@ -1482,6 +1482,13 @@ protected MrpLine createMrpLine(
             I18n.get(IExceptionMessage.MRP_NO_PRODUCT_UNIT),
             product.getFullName());
       }
+      if (product.getId() == null) {
```

Comment:
Exception should be throw if product.getId() is not defined in the  this.productMap because product.getId will never return a NPE.